python/dataset.cpp

```diff
@@ -78,6 +78,8 @@ void bind_dataset_proxy_methods(py::class_<T, Ignored...> &c) {
   c.def("__contains__", &T::contains);
   c.def("copy", [](const T &self) { return Dataset(self); },
         "Return a (deep) copy.");
+  c.def("sum", [](const T &self, const Dim &dim) { return sum(self, dim); });
+  c.def("mean", [](const T &self, const Dim &dim) { return mean(self, dim); });
```

Comment:
We should be consistent here and make these free functions, not methods, just like for all the others like `histogram` and `merge`. I see you have those below, so at least for now I would avoid adding unnecessary methods.